pkg/kube/krt/mergejoin.go

```diff
@@ -313,3 +356,86 @@ func (j *mergejoin[T]) updateIndexLocked(e Event[T], key Key[T]) {
 		}
 	}
 }
+
+func (j *mergejoin[T]) runQueue(initialCollections []Collection[T], subscriptionFunc func([]Event[T])) {
+	// Now that we've subscribed, process the current set of collections.
+	regs := []HandlerRegistration{}
+	for _, c := range initialCollections {
+		// Ensure each sub-collection is synced before we're marked as synced.
```

Comment:
This is checking to make sure the event handlers have synced; not the actual collections. This doesn't do a check on initialCollections[].hasSynced, so you probably want to do that first before registering event handlers